<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>settingsDlg</class>
 <widget class="QDialog" name="settingsDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>520</width>
    <height>354</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>520</width>
    <height>246</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>520</width>
    <height>400</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>520</width>
    <height>246</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>310</y>
     <width>501</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Reset|QDialogButtonBox::Save</set>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>501</width>
     <height>281</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="settingsLayout">
    <item row="0" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QLineEdit" name="sync_interval">
        <property name="toolTip">
         <string>How frequently the tool will scan the data folder</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <item>
       <widget class="QLineEdit" name="ftp_pass">
        <property name="toolTip">
         <string>FTP password</string>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Remote User</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <widget class="QLineEdit" name="ftp_user">
        <property name="toolTip">
         <string>FTP user name</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Remote Host</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="ftp_host">
      <property name="toolTip">
       <string>IP address or the url where the files to be sent</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QTimeEdit" name="house_keeping">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When would the housekeeping start. &lt;/p&gt;&lt;p&gt;When it starts, files added afterwards will not be transferred to the client site.&lt;/p&gt;&lt;p&gt;Once all the files in queue are processed, a backup of transferred files will be created at the place indicated by the 'Backup Path' in a dated folder.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="displayFormat">
           <string>h:mm AP</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>268</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="9" column="0">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>Backup Path</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>FTP Daemons</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_9">
      <property name="text">
       <string>FTP Mode</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>House keeping start at</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Folder Scan interval (sec)</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Remote Password</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Log Path</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLineEdit" name="bkup_path">
        <property name="toolTip">
         <string>Path where the transferred files will be archieved.</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_setBkupPath">
        <property name="text">
         <string>Set</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="8" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLineEdit" name="log_path">
        <property name="toolTip">
         <string>Path where the log will be written</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_setLogPath">
        <property name="text">
         <string>Set</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="5" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QComboBox" name="ftp_mode">
        <property name="toolTip">
         <string>FTP mode of transfer [Dependant on FTP and Firewall setup]</string>
        </property>
        <property name="currentText">
         <string/>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="maxVisibleItems">
         <number>2</number>
        </property>
        <property name="maxCount">
         <number>2</number>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::NoInsert</enum>
        </property>
        <property name="duplicatesEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>78</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="7" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <widget class="QLineEdit" name="thread_count">
        <property name="toolTip">
         <string>This is a future enhancement.</string>
        </property>
        <property name="text">
         <string>1</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>288</width>
          <height>17</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_10">
      <property name="text">
       <string>Remote Path</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLineEdit" name="ftp_cd_path">
      <property name="toolTip">
       <string>FTP password</string>
      </property>
      <property name="echoMode">
       <enum>QLineEdit::Normal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>501</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <tabstops>
  <tabstop>sync_interval</tabstop>
  <tabstop>ftp_host</tabstop>
  <tabstop>ftp_user</tabstop>
  <tabstop>ftp_pass</tabstop>
  <tabstop>ftp_cd_path</tabstop>
  <tabstop>ftp_mode</tabstop>
  <tabstop>house_keeping</tabstop>
  <tabstop>thread_count</tabstop>
  <tabstop>log_path</tabstop>
  <tabstop>btn_setLogPath</tabstop>
  <tabstop>bkup_path</tabstop>
  <tabstop>btn_setBkupPath</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>btn_setLogPath</sender>
   <signal>clicked()</signal>
   <receiver>settingsDlg</receiver>
   <slot>setLogPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>477</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>277</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>settingsDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>settingsDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>settingsDlg</receiver>
   <slot>onActionBtnClick(QAbstractButton*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_setBkupPath</sender>
   <signal>clicked()</signal>
   <receiver>settingsDlg</receiver>
   <slot>setBkupPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setLogPath()</slot>
  <slot>checkSettings()</slot>
  <slot>setSyncDir()</slot>
  <slot>syncTimeChanged()</slot>
  <slot>onActionBtnClick(QAbstractButton*)</slot>
  <slot>setBkupPath()</slot>
 </slots>
</ui>
